centreon/src/Core/Contact/Domain/AdminResolver.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Core\Contact\Domain;
+
+use Centreon\Domain\Contact\Interfaces\ContactInterface;
+use Core\Security\AccessGroup\Application\Repository\ReadAccessGroupRepositoryInterface;
+use Core\Security\AccessGroup\Domain\Model\AccessGroup;
+
+/**
+ * This service allows to determine if a contact is admin or not.
+ * It would probably be better to add this logic to the Contact model but
+ * for now we keep it that way to avoid touching too many things.
+ */
+final readonly class AdminResolver
+{
+    public function __construct(
+        private readonly ReadAccessGroupRepositoryInterface $readAccessGroupRepository,
+        private readonly bool $isCloudPlatform,
```

Comment:
The class is readonly
```suggestion
        private ReadAccessGroupRepositoryInterface $readAccessGroupRepository,
        private bool $isCloudPlatform,
```

Reply:
I had to make some changes because of pest.
It's no longer readonly and no longer final. 
I don't like it but otherwise I'm forced to add an interface which would be acceptable at worst...

Reply:
It's ok. Thanks.